client/tsconfig.json

```diff
@@ -6,9 +6,37 @@
   "files": [
     "../package.json",
     "../server/types/socket-events.d.ts",
+    "../server/helper.ts",
+    "../server/log.ts",
+    "../server/config.ts",
+    "../server/client.ts",
+    "../server/clientManager.ts",
+    "../server/identification.ts",
+    "../server/plugins/changelog.ts",
+    "../server/plugins/uploader.ts",
+    "../server/plugins/storage.ts",
+    "../server/plugins/inputs/index.ts",
+    "../server/plugins/messageStorage/sqlite.ts",
+    "../server/plugins/messageStorage/text.ts",
+    "../server/plugins/packages/index.ts",
+    "../server/plugins/packages/publicClient.ts",
+    "../server/plugins/packages/themes.ts",
+    "../server/plugins/webpush.ts",
+    "../server/plugins/sts.ts",
+    "../server/plugins/clientCertificate.ts",
+    "../server/plugins/auth.ts",
+    "../server/plugins/auth/local.ts",
+    "../server/plugins/auth/ldap.ts",
+    "../server/plugins/irc-events/link.ts",
+    "../server/command-line/utils.ts",
+    "../server/models/network.ts",
+    "../server/models/user.ts",
+    "../server/models/msg.ts",
+    "../server/models/prefix.ts",
     "./js/helpers/fullnamemap.json",
```

Comment:
this works, however that includes way more than I expected.
Why do we need to include files which are purely server side? sqlite for instance is never actually imported in the client.

Is this due to transitive deps from some of the actual necessary imports or are we compiling the whole world and not just the client?

(Generally curious, I assume you had your reasons, so don't take this as a complaint)

Reply:
Of course, that's alright! 

And correct, it's due to the transitive deps from code sharing. This is all a result of the dependency import tree being a bit wonky between client/server side code sharing. Each of these were an import used in the TS project (and the compiler would fail if they weren't provided to tsconfig). I'm going to try and move the code around to limit this list (and therefore fix some of the dynamic import errors).